src/Orc.Controls/Orc.Controls.Shared/Controls/DateTimePicker/Views/DateTimePickerControl.xaml.cs

```diff
@@ -45,18 +50,36 @@ public DateTimePickerControl()
                 NumericTBMinute,
                 NumericTBSecond,
             };
+        }
 
-            NumericTBDay.RightBoundReached += NumericTextBoxOnRightBoundReached;
-            NumericTBMonth.RightBoundReached += NumericTextBoxOnRightBoundReached;
-            NumericTBYear.RightBoundReached += NumericTextBoxOnRightBoundReached;
-            NumericTBHour.RightBoundReached += NumericTextBoxOnRightBoundReached;
-            NumericTBMinute.RightBoundReached += NumericTextBoxOnRightBoundReached;
+        private void SubscribeNumericTextBoxes()
+        {
+            _numericTextBoxes[0].RightBoundReached += NumericTextBoxOnRightBoundReached;
```

Comment:
Yes, I agree this is weird, I was confused too.
Each field (year, month, day) for date is numericTextBox, when you are inside field you can press left or right keyboard button, if you are at first or last position in numericTextBox (which is TextBox) then left or right button switch you to next numericTextBox, so 0-4 because we can't move right in last (5 index) and 5-1  because we can't move  left in first (0 index) numerictextbox. I hope you see it :)